packages/yoga/src/Input/web/Input.jsx

```diff
@@ -160,6 +172,9 @@ const Input = React.forwardRef(
               value={value}
               onChange={onChange}
               {...a11yFieldProps}
+              style={
+                prependElementWidth ? { textIndent: prependElementWidth } : {}
```

Comment:
This should not be needed if the input uses `flexbox` or `grid` code. Just set a `gap` between the elements would work :)

Reply:
I don't get how this would work @MicaelRodrigues 
The element to be rendered is inside another one, the PreprendElement, that has an absolute position (the same approach used for the right icon).

<img width="274" alt="image" src="https://github.com/gympass/yoga/assets/135242379/3607a6ff-6276-473e-8489-81cf632740b5">
<img width="403" alt="image" src="https://github.com/gympass/yoga/assets/135242379/a87033af-da05-4958-b89e-17baab8c1d71">

That's the reason the Field has an indent.

To use the flexbox, the parent component (Fieldset) would need to be changed but this can lead to some unhandled scenarios.

Reply:
> To use the flexbox, the parent component (Fieldset) would need to be changed but this can lead to some unhandled scenarios.

I'd prefer to change the code than to introduce a prop the forces the user to define a width of an element to indent the text... It's not very responsive. After all, it's just css. 
> I don't get how this would work @MicaelRodrigues
I think `PhoneInput` has an implementation like the one i'm suggesting.  But, I can try to find a time to help you when i got back to work next week :)

Reply:
My main concern in changing the code to use flexbox is that everything is absolute positioned inside the Input component.
The input, the label, in the DOM:
![image](https://github.com/gympass/yoga/assets/135242379/0b603e2a-6922-4a42-a96d-da2cce9d55ae)

the styles:
![image](https://github.com/gympass/yoga/assets/135242379/50169523-8713-45f9-af47-cd1adc1acfb3)
from the input

![image](https://github.com/gympass/yoga/assets/135242379/4487f4e8-ddb7-4aba-b205-7eee1c8d3b5e)
from the label

I guess those elements are absolute to force their positions inside the parent one, the Fieldset. I don't see an _easy_ way to refactor this only to replicate a similar UI from the right positioned icon.


About the width, I guess the UI won't break, because the current Input behavior doesn't change. If there is no leftElement prop, nothing changes. And if a parent component passes this prop, it is its responsibility to know the width of the element it is passing to a child component. In other words, the parent component must know if it is rendering an icon width 20ox of width or a credit card flag with 48px.